solr/core/src/java/org/apache/solr/util/tracing/TraceUtils.java

```diff
@@ -32,7 +33,7 @@ public static void setDbInstance(SolrQueryRequest req, String coreOrColl) {
   }
 
   public static void ifNotNoop(Span span, Consumer<Span> consumer) {
-    if (span != null && !(span instanceof NoopSpan)) {
+    if (span != null && !(span instanceof NoopSpan) && !(span instanceof SimplePropagatorSpan)) {
```

Comment:
Arguably, SimplePropagatorSpan should implement NoopSpan?

Reply:
ugh, of course, good one